src/extension/src/controller.ts

```diff
@@ -66,8 +71,25 @@ export class Controller {
       vscode.window.showErrorMessage(CONSTANTS.ERRORS.INVALID_MODULE);
     }
   }
+  /**
+   * Provides access to the Controller. Maintains Singleton Pattern. Function will bring up ReactPanel to View if Controller instance exists, otherwise will instantiate a new Controller.
+   * @param message The payload received from the wizard client. Message payload must include field 'module'
+   * @returns Singleton Controller type
+   */
+  public static getInstance(
+    context: vscode.ExtensionContext,
+    extensionStartTime: number
+  ) {
+    if (this._instance) {
+      this._instance.showReactPanel();
+    }
```

Comment:
NIT: would it be more clear to have `else { this._instance = ... }` here instead of as an or statement below?

Reply:
I agree it is es6 magic... your call

Reply:
Yeah...you can leave it like this, it is not a big dieal

Reply:
I think having an else is a lot more clearer because it is a little bit confusing

Reply:
we will do it in a separate PR